### ISSUE TITLE
explain: WTF use Ionic?

### ISSUE THREAD

**Issue by dpcunningham** [dpcunningham/process-spinup-devenv-ionic4-ng-on-linux#2]
...because all the kool kids are using X or Y or Z, _right?_

We are a startup.  Funds are limited.  Maybe... _maybe_ when we have oodles of money,  we'll do  all the stuffs necessary for parallel development in Swift for iOS, or Java for Android.  Or was that C++?  Or Kotlin?  _Hmmmm...._

Single code base.  Multiple platforms.  Do I need to say more?

_But single code bases never work!_  Bullshit.  Back in paleolithic times, I wrote a cross platform charting and graphing component library, in C++, that ran on Windows, OS2, Mac, Next,  X, and even `curses`.  Yes, that's ancient tech, and (IMO) strongly-typed, but the principles of CS, and making reusable software don't change. 

Performance?  Not as fast?  Yup.  Don't care.  For now.  The features we are deploying are generic, and not dependent on ultra-render speed.  In fact, we are going to aim for simpler and simpler and simpler UX experiences.

Guess what?   _We don't even know what the best UX is!_   This is something that will only be _discovered in partnership with our customers_.  It will  be in constant flux for the foreseeable future.  So again, reducing the number of moving parts by leveraging a single, rapidly-adapting code base is my dictum as CTO, until I fire myself (because I have found someone better).

My recommendation:  If you can't get past this, then you should not be involved in our project.

More pros and cons are hashed out  below...

**Comment by dpcunningham** [comment]
Just to be fair, let's hear from the other sides...

### Ionic vs _Whatever_...
        
- [Xamarin or Ionic, which one to choose? (infographic)](https://www.valuecoders.com/blog/technology-and-apps/xamarin-ionic-choose-one/#)

- [React Native vs. Xamarin vs. Ionic vs. Flutter: Which is better?](https://www.apptunix.com/blog/frameworks-cross-platform-mobile-app-development/)

- From [Battle of the Frameworks: Ionic vs Xamarin](https://smartbridge.com/ionic-vs-xamarin/) by Smartbridge

    - Hybrid HTML frameworks use web technologies JavaScript/HTML/CSS and are interpreted and displayed within the device’s native web view as a standalone application.Cross-Platform Native frameworks use one programming language across multiple platforms to develop and compile straight into an intermediate language.Based on these definitions, we can now begin to understand why Ionic, being Hybrid HTML, and Xamarin, being a Cross-Platform Native, perform, look and feel the way they do.

    - Considering that the Ionic platform uses web technologies to produce a hybrid application, it will only offer an average user experience and pale in comparison to Xamarin apps. Xamarin will offer the closest experience to native applications and will be the tool of choice if you’re looking to compete with other applications on the market. Below are analytics that highlight the performances of Hybrid HTML mobile apps versus Cross-Platform Native apps [...]


Bottom line?   _Yeah,  yeah,  Ionic sucks._  

Thanks.  Moving on...
